bpf/lib/host_firewall.h

```diff
@@ -394,7 +437,8 @@ ipv4_host_policy_egress(struct __ctx_buff *ctx, __u32 src_id,
 	if (ct_buffer.ret < 0)
 		return ct_buffer.ret;
 
-	return __ipv4_host_policy_egress(ctx, src_id == HOST_ID, ip4, &ct_buffer, trace, ext_err);
+	return __ipv4_host_policy_egress(ctx, src_id == HOST_ID || ipcache_srcid == HOST_ID,
+					ip4, &ct_buffer, trace, ext_err);
```

Comment:
Let's please tackle the two problems with separate patches / PRs. Understanding the IPv6 whitelisting change is difficult enough, this subtle IPv4 change only makes things more complicated. And it introduces a severe regression for current users (they now suddenly need host-level IPv4 egress policies for their SNATed pod traffic).

So let's focus this first patch on the missing IPv6 allow-listing, when combining HostFW with kube-proxy based masquerading. Could you update the PR / patch description accordingly? Happy to help with tackling `--install-iptables-rules=false` once this is sorted.

Reply:
Understood, will make the changes.